DBZ-2127 Re-unify content for the MySQL connector in one file

TovaCohen opened the pull request:
[DBZ-2127](https://issues.redhat.com/browse/DBZ-2127) requires putting all the content for the MySQL connector back into one file: `mysql.adoc`

However, individual assembly and module files are still needed downstream. To meet this requirement, there are now comments at the beginning of sections. These comments specify annotations that we use to split the files into assemblies and modules when we fetch the upstream content and reuse it downstream. 

I ran Antora locally to confirm that this content builds correctly. I compared the content in the reunified file with the content that we publish on the community site. I am confident that I did not leave out any content, but this was a manual process and my work needs to be checked. 

I deleted the files that were in the `partials/assemblies/mysql-connector` and `partials/modules/mysql-connector` directories and I deleted those directories. 

I searched for each instance of an anchor ID that is no longer used and I made sure that deleted anchor IDs would not cause errors. I updated references to anchor IDs where needed. 

I edited the content to conform with Red Hat style conventions. I did some reorganizing. Eventually, the organization of the content for each connector will be the same, as much as possible. This is an incremental process. The content organization for the MySQL connector closely mimics the organization of the content for the Db2 connector. 

Manually (not using our fetch script), I copied the reunified mysql.adoc file downstream, and I ran the commands to split it. With a little tweaking, the downstream user guides builds and correctly renders the content. But, I will not know for sure that this work is complete until I can run the fetch script, build the fetched content, and render the content in Pantheon.  So more updates might be needed after I do those tasks. In other words, please do NOT close DBZ-2127 after this PR is merged. I need to keep it open until I do the downstream work, and I expect that the downstream work will require some additional small edits.

TovaCohen commented:
Just want to note that the commit for "Another issue identified...." is an update for [DBZ-2610](https://issues.redhat.com/browse/DBZ-2610), which reports a missing content directory specification in the command that builds the container for deploying the connector. I was afraid of introducing a conflict if I did the MySQL update as part of the PR for the updates to the doc for the other connectors.

Naros commented on documentation/modules/ROOT/pages/connectors/mysql.adoc:
```diff
@@ -6,20 +9,2563 @@
+
+The value portion of a change event for a change to this table is described for: 
+
+* <<mysql-create-events,_create_ events>>
```
I'm of the opinion that now that we've added Tombstone and PK updates options here, should we consider formatting these entries despite how the headings are may formatted?  To me this goes back to having camel-case elements and upper-case elements when we discussed `HSTORE` and `TIMESTAMP` in the PG docs.

TovaCohen replied:
Sorry. I don't understand what you want to see in the doc.
The way it is now, the user will see these links: 

_create_ events
_update_ events
Primary key updates
_delete_ events
Tombstone events

How should this change?

Naros replied:
I wonder if it would look more uniform if the text here was

Create events
Update events
Primary key updates
Delete events
Tombstone events

gunnarmorling replied:
Agreed, the italic setting looks a bit odd when looking at it in the navigation menu.

TovaCohen replied:
I feel strongly that the headings need to immediately convey, as much as possible, that "create", "delete", and "update" are being used as adjectives and not as imperative verbs. While yes, the headings you suggest, Chris, are uniform, they are less helpful to a user who is trying to learn. 

We already made the update I am suggesting in the doc for Db2 and for PostgreSQL. Whatever we decide here, will need to be applied to the doc for those connectors, and to the doc for the other connectors as we update their doc. You can see how it looks in the navigation panel now: 
https://debezium.io/documentation/reference/1.2/connectors/postgresql.html#postgresql-events
https://debezium.io/documentation/reference/1.2/connectors/db2.html#db2-events 

Please reconsider leaving it the way I did it. 
Alternatively, I would be okay (though not as happy) with:

CREATE events
UPDATE events
Primary key updates
DELETE events
Tombstone events

Naros commented on documentation/modules/ROOT/pages/connectors/mysql.adoc:
```diff
@@ -6,20 +9,2563 @@
+|===
+
+[id="mysql-advanced-connector-configuration-properties"]
+.Advanced MySQL connector configuration properties
```
I am curious if rendering of this type of layout needs some CSS changes upstream.  What I actually see upstream when this gets rendered is the following:

```
Advanced MySQL connector configuration properties
The following table describes ...
```

You'll notice there is no spacing between the heading, no underline which essentially makes the heading flow into the text so you might actually miss notion that the section has changed to something else.  I've added [DBZ-2631](https://issues.redhat.com/browse/DBZ-2631) as a follow-up for this.

Naros replied:
DBZ-2631 has been resolved.

TovaCohen replied:
Oh, great! I meant to bring this up. I don't see how you changed the rendering, but yes, we needed to somehow distinguish the 

`.This is a section title `

headings. So thanks for taking care of that!

Naros replied:
This is handled automatically by our CSS used by Antora in our website build process; so this is entirely transparent to your PR and only deals specifically with the upstream HTML generation step.

Naros reviewed (changes requested):
Hi @TovaCohen, great work on reunifying this.  I've found a few things which I've highlighted below.  

As an aside, one common theme I noticed throughout is that the anchors that were used in the prior split docs have changed in various spots.  Since these changes only apply to 1.4, it is something we can likely accept but going forward we really should try and keep anchors as static as possible.  This just helps users who bookmark parts of the documentation keep those bookmarks valid.

TovaCohen commented on documentation/modules/ROOT/partials/assemblies/mysql-connector/assembly-overview-of-how-the-mysql-connector-works.adoc:
```diff
@@ -1,25 +0,0 @@
-[id="overview-of-how-the-mysql-connector-works"]
-= Overview of how the MySQL connector works
-
-toc::[]
```
This is back in place with the next commit.

TovaCohen commented on documentation/modules/ROOT/pages/connectors/mysql.adoc:
```diff
@@ -6,20 +9,2563 @@
+[[how-the-mysql-connector-works]]
+== How the connector works
+
+To optimally configure and run a {prodname} MySQL connector, it is helpful to understand how the connector tracks the structure of tables, exposes schema changes, performs snapshots, and determines Kafka topic names. An overview of the MySQL topologies that the connector supports is useful for planning your application. 
```
I moved that last sentence in the introductory paragraph, about topologies, to be the first sentence. 
I moved the toplogies section to be the first subsection in How the connector works. 
I don't think the list of xref links is needed in the text panel because they will appear in the navigation panel on the right upstream.  Downstream, you see the links for the subsections only if you clicked the main section that contains them, and that's why I add them for downstream rendering. In other words, downstream, if you are scrolling content, the navigation panel on the left does not necessarily show you what you are looking at, and it is very long because it is for the whole book. Let me know if you want something different. Otherwise, this update will be in the next commit.

TovaCohen commented:
Chris, I made most of your updates with the most recent commit. There's one outstanding question about the subheadings in the change event descriptions section. 
I also moved the two lines that declare the file's toplevel anchor ID and heading to be just below the first two lines in the file, which are comments, that is, annotations for splitting the files. This is needed for the nebel utility to correctly split the file. 
As for the old anchor IDs, I'll remember to keep them in place, just in case. Right now, there is a bug, or it might be a limitation, of the split tool, that there cannot be two consecutive anchor ID declarations with this format: 

```
[[first-anchor-id]]
[[next-anchor-id]]
```

I'll talk to Fintan about this and hopefully it is something he can fix. But that's why I deleted the ones that at least had no references in the doc set.

gunnarmorling commented:
Hey @TovaCohen, @Naros, where are we with this one? Anything left to be discussed? Always feeling a bit uneasy with such large PRs pending for a longer time due to the potential merge headaches they may cause. Thx!

TovaCohen commented:
@gunnarmorling  and @Naros -- From my perspective, if you are will to concede that the headings for the descriptions of the different event types can be italic, then this is done. I've made the other changes that were needed. If not, then one more update is needed. Your call. See the previous comment about the headings.

gunnarmorling commented:
I'm fine personally with italic. I'd dislike upper-case, as this might suggest a relationship to SQL keywords to someone, whereas this should be more about the semantics of the different event types here. @Naros?

Naros commented:
Sounds like we've reached a decision, I'll give this one quick final review & merge.
